### PR TITLE
KAFKA-7455: Support JmxTool to connect to a secured RMI port.

### PR DESCRIPTION
**Pull request by murong00**
To connect to a secured RMI port (enabling remote JMX with password authentication and SSL), JmxTool should pass an envionrment map that contains relevant certification entry.

### CODE REVIEW

File: core/src/main/scala/kafka/tools/JmxTool.scala [hunk 1]
```diff
@@ -117,7 +131,19 @@ object JmxTool extends Logging {
...
+        val env = new java.util.HashMap[String, AnyRef]
+        // ssl enable
+        if (enableSsl) {
+          val csf = new SslRMIClientSocketFactory
+          env.put("com.sun.jndi.rmi.factory.socket", csf)
+        }
+        // password authentication enable
+        if (enablePasswordAuth) {
+          val userPassword = options.valueOf(jmxAuthPropOpt).split("=", 2)
+          val credentials = Array(userPassword(0), userPassword(1))
```
Comment (asasvari) [inline core/src/main/scala/kafka/tools/JmxTool.scala]: userPassword is not needed, you can have something like
```
val credentials = options.valueOf(jmxAuthPropOpt).split("=", 2)
```

**Review by asasvari — COMMENTED** [review]
@murong00 Many thanks! I tested that JmxTool now can connect to SSL-enabled JMX port. I only left some minor comments.

@omkreddy can you take a look?

File: core/src/main/scala/kafka/tools/JmxTool.scala (hunk 1 above)
Comment (murong00) [reply core/src/main/scala/kafka/tools/JmxTool.scala]: Changed, thanks!

**Comment by murong00** [comment]
@omkreddy PTAL, thanks!

**Comment by omkreddy** [comment]
This may require a minor KIP.  
@ijuma What do you think?

**Comment by ijuma** [comment]
Yes, that's correct.

**Comment by murong00** [comment]
@omkreddy See more details in [KIP-417](https://cwiki.apache.org/confluence/display/KAFKA/KIP-417%3A+Allow+JmxTool+to+connect+to+a+secured+RMI+port), thanks!

**Comment by tal705** [comment]
@omkreddy  Are there any plans to have this pull request being merged soon?

**Comment by murong00** [comment]
retest this please

**Comment by murong00** [comment]
@ijuma @omkreddy Could you help to review this minor improvement when you are available, thanks!

**Comment by omkreddy** [comment]
@murong00 Thanks for the PR. LGTM. Can you bump up the KIP-417 vote thread? We need one more binding vote to pass the KIP.

**Comment by murong00** [comment]
@omkreddy Binding vote via [this thread](https://lists.apache.org/thread.html/313c7680fc7fa2a62ac413c649c1758dfabbd671366ea54ef4056f7f@%3Cdev.kafka.apache.org%3E), but I'm not sure whether it works.

**Comment by omkreddy** [comment]
retest this please

**Comment by omkreddy** [comment]
@murong00 Sorry for the delay. Merging to trunk.

**Pull request merged**